Gem/Code/Source/MultiplayerSampleSystemComponent.cpp

```diff
@@ -108,15 +108,16 @@ namespace MultiplayerSample
         return AZ::TICK_PLACEMENT + 2;
     }
 
-    Multiplayer::NetworkEntityHandle MultiplayerSampleSystemComponent::OnPlayerJoin(
-        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const Multiplayer::MultiplayerAgentDatum& agentDatum)
+    Multiplayer::NetworkEntityHandle MultiplayerSampleSystemComponent::OnPlayerJoin([[maybe_unused]] uint64_t userId, [[maybe_unused]] const Multiplayer::MultiplayerAgentDatum& agentDatum, EntityPreActivationCallback preActivationCallback)
     {
         AZStd::pair<Multiplayer::PrefabEntityId, AZ::Transform> entityParams = AZ::Interface<IPlayerSpawner>::Get()->GetNextPlayerSpawn();
 
         Multiplayer::INetworkEntityManager::EntityList entityList =
```

Comment:
This feels like this could be collapsed in MultiplayerSystemComponent rather than a callback considering the callback is calling EnableAutotnomousControl. Could CreateEntitiesImmediate be handling the EnableAutonomousControl? Alternatively, could there be an IMultiplayer call added that handles EnableAutonomousControl?

Reply:
Cleaned this up, let me know what you think! We are really bound to having autonomy enabled _before_ entity activation. I ended up moving the enabling of autonomous control into NetworkEntityManager before the entities are activated; also added a helper method "CreateAutonomousPlayerImmediate" (a nicer/simpler method for IMultiplayerSpawner to call with less parameters). This way, each project's MultiplayerSpawner could go to town modifying the entities before passing it back to MultiplayerSystemComponent, and still know everything will just work.